File: LeetCode/2002/main.cpp
class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& arr, int m, int n) {
        
        if(arr.size() != m * n) {
            return vector<vector<int>> ();
        }

        vector<vector<int>> mat(m, vector<int> (n, 0));

        int row = 0;
        for(int i = 0; i < arr.size(); i++) {
            if(i != 0 && i % n == 0) {
                row++;
            }
            mat[row][i % n] = arr[i];
        }
        
        return mat;

    }
};

File: LeetCode/2002/main.py
class Solution:
    def construct2DArray(self, arr: List[int], m: int, n: int) -> List[List[int]]:

        if (len(arr) != m * n):
            return []

        mat = [[0 for i in range(n)] for j in range(m)]
        row = 0

        for i in range(len(arr)):
            if (i != 0 and i % n == 0):
                row += 1

            mat[row][i % n] = arr[i]

        return mat
